src/resolver/generics.rs

```diff
@@ -92,14 +90,21 @@ impl<'i> TypeAnnotator<'i> {
                     if let Some(pou) = self.index.find_pou(qualified_name) {
                         //only register concrete typed function if it was not indexed yet
                         if self.index.find_pou(new_name.as_str()).is_none() {
-                            //register the pou-entry, implementation and member-variables for the requested (typed) implemmentation
-                            // e.g. call to generic_foo(aInt)
-                            self.register_generic_pou_entries(
-                                pou,
-                                cloned_return_type.as_str(),
-                                new_name.as_str(),
-                                generic_map,
-                            );
+                            if let StatementAnnotation::Function { return_type, .. } = &annotation {
```

Comment:
this looks like `return_type` may be different from the original return type from line 72?
No tests fail if i switch line 98 back to `cloned_return_type`